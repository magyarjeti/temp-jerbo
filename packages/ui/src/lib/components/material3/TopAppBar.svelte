<script lang="ts">
	import { concatClasses } from '@packages/toolbox/concatClasses';
	import type { Snippet } from 'svelte';

	interface Props {
		children?: Snippet;
		onstuck?: (isStuck: boolean) => void;
		class?: string | null;
	}

	let { children, onstuck, ...props }: Props = $props();

	let isStuck = $state(false);
	let el: HTMLElement | undefined = $state();

	function setSticky() {
		if (el) {
			const rect = el.getBoundingClientRect();
			let stuckResult = false;
			if (window.scrollY === 0) {
				stuckResult = false;
			} else if (window.scrollY > 0 && rect.top + window.scrollY <= window.scrollY) {
				stuckResult = true;
			} else {
				stuckResult = false;
			}
			isStuck = stuckResult;
			onstuck?.(stuckResult);
		}
	}
</script>

<svelte:window onscroll={setSticky} />

<nav
	class={props.class ||
		concatClasses(
			'sticky top-0 z-10 h-12 gap-2 bg-surface text-on-surface transition-shadow',
			isStuck && 'elevation-2',
		)}
	bind:this={el}
>
	{@render children?.()}
</nav>
