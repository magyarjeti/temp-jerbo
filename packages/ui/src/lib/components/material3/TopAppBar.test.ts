import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import TopAppBar from './TopAppBar.svelte';

test('TopAppBar snapshot', () => {
	const { container } = render(TopAppBar);
	expect(container).toMatchSnapshot();
});
