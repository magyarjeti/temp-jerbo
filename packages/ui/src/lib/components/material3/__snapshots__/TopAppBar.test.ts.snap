// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopAppBar snapshot 1`] = `
<div>
  <nav
    class="bg-surface text-on-surface sticky top-0 z-10 h-12 gap-2 transition-shadow"
  >
    <div
      class="h-full"
    >
      <div
        class="flex h-full items-center justify-between gap-4"
      >
        <div>
          <button
            class="cursor-pointer inline-flex items-center justify-center shape-full typo-label-medium h-10 w-10 text-[1em] text-primary state-layers state-layer-hover-on-surface-variant state-layer-focus-on-surface-variant disabled:text-on-surface/38"
          >
            <span
              class="jeti-icon svelte-1pzkbd4"
            >
              menu
            </span>
          </button>
          
        </div>
         
      </div>
    </div>
  </nav>
</div>
`;
